### ISSUE TITLE
Local version with local gallery

### ISSUE THREAD

**Issue by nar001** [thefapinstructor/fapinstructor#31]
Since the website is pretty well done already, having a local version with a possibility to use a local gallery could be pretty nice, if possible! It would certainly be easier planning a session since you wouldn't have to hope the tumblr or subreddit has good stuff!

**Comment by The1nstructor** [comment]
that was fast 😆

**Comment by nar001** [comment]
I received the notification, and since I wasn't doing anything special I thought I'd open it now! :p

**Comment by The1nstructor** [comment]
@nar001 

>  It would certainly be easier planning a session since you wouldn't have to hope the tumblr or subreddit has good stuff!

Yes! And It would make it possible to even play offline 😄 

> I received the notification, and since I wasn't doing anything special I thought I'd open it now! :p

Very good idea. Putting stuff here makes it easier to track and to work on 😃

**Comment by nar001** [comment]
Yeah, I was using edgemeplease, but your website is still being updated and just easier to use, so it was nice when I found it! Lots more options too!

**Comment by The1nstructor** [comment]
it's not mine ;P its @thefapinstructor's 

sorry for the name confusion ;) 

And I think so, too. So I decided to do also some coding here. I really like the integration of tumblrs and reddits in the website. And the abillty to configure it. Actually I'm planning on some similar functionalities like in edgemeplease. #27 😉

**Comment by nar001** [comment]
Woops, misread your name! And thanks for helping work on it too! And which functionalities are you thinking of, anything specific yet?

**Comment by The1nstructor** [comment]
Just browse through the issues 😆 or the pull requests.

Currently I am testing some new nipple play cards I created the past days.

**Comment by ArtisticMenase** [comment]
The main problem here is getting the app to recognise its being run locally and actually accessing a local source.

Potentially, we could access build environment variables to see if we're running development/local and then the user would have to set a relative location for images, not sure how that retrieval would work for getting a single image though.

At least part of this (from my little amount of research) is possible, it's the local file access I'm unsure about.

**Comment by The1nstructor** [comment]
I see the problem. 

Currently, we are only working on the website hosted version. Only if the user uses git clone and installs the game on his machine, a local version of the game is already available. 

So the specification of a path where to read the offline images would definitely raise a new security discussion. 

In my opinion, focussing on a new *remote* version is not essential at the moment.

**Comment by edginggames** [comment]
I agree that just downloading the source and running it would work just fine.  Could it work remotely if you used the file://  protocol?  You could have a file selection box to point at a folder and then recursively crawl the dir for valid media.

**Comment by The1nstructor** [comment]
And we would have to disable // operations for security reasons.

**Comment by edginggames** [comment]
could we use something like: https://www.html5rocks.com/en/tutorials/file/dndfiles/
